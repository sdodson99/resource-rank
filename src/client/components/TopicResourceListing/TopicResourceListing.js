import React from 'react';
import PropTypes from 'prop-types';
import styles from './TopicResourceListing.module.css';
import TopicResourceListingItem from '../TopicResourceListingItem/TopicResourceListingItem';
import PaginatedListing from '../PaginatedListing/PaginatedListing';

const TopicResourceListing = ({
  topicId,
  topicSlug,
  topicResources,
  selectedPage,
  pageCount,
  onPageClick,
}) => {
  const topicResourcesListingItems = topicResources.map((r) => (
    <TopicResourceListingItem
      key={r.resource.id}
      topicId={topicId}
      topicSlug={topicSlug}
      resourceId={r.resource.id}
      resourceSlug={r.resource.slug}
      name={r.resource.name}
      rating={r.ratingList.average}
      verified={r.resource.verified}
    />
  ));

  return (
    <div
      className={styles.TopicResourceListing}
      data-testid="TopicResourceListing"
    >
      <PaginatedListing
        selectedPage={selectedPage}
        pageCount={pageCount}
        onPageClick={onPageClick}
      >
        {topicResourcesListingItems}
      </PaginatedListing>
    </div>
  );
};

TopicResourceListing.propTypes = {
  topicId: PropTypes.string,
  topicSlug: PropTypes.string,
  topicResources: PropTypes.arrayOf(
    PropTypes.shape({
      resource: PropTypes.shape({
        id: PropTypes.string,
        slug: PropTypes.string,
        name: PropTypes.string,
        verified: PropTypes.bool,
      }),
      ratingList: PropTypes.shape({
        average: PropTypes.number,
      }),
    })
  ),
  selectedPage: PropTypes.number,
  pageCount: PropTypes.number,
  onPageClick: PropTypes.func,
};

TopicResourceListing.defaultProps = {
  topicResources: [],
};

export default TopicResourceListing;
