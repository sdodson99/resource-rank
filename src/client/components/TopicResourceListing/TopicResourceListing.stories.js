import React from 'react';
import TopicResourceListing from './TopicResourceListing';

export default {
  component: TopicResourceListing,
  title: 'Components/TopicResourceListing',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <TopicResourceListing {...args} />;

export const Default = Template.bind({});
Default.args = {
  topicId: '1',
  topicSlug: 'topic-1',
  topicResources: [
    {
      resource: {
        id: '1',
        slug: 'resource-1',
        name: 'Resource 1',
        verified: true,
      },
      ratingList: {
        average: 1,
      },
    },
    {
      resource: {
        id: '2',
        slug: 'resource-2',
        name: 'Resource 2',
      },
      ratingList: {
        average: 3,
      },
    },
    {
      resource: {
        id: '3',
        slug: 'resource-3',
        name: 'Resource 3',
      },
      ratingList: {
        average: 5,
      },
    },
  ],
  selectedPage: 3,
  pageCount: 5,
};

export const Single = Template.bind({});
Single.args = {
  topicId: '1',
  topicSlug: 'topic-1',
  topicResources: [
    {
      resource: {
        id: '1',
        slug: 'resource-1',
        name: 'Resource 1',
      },
      ratingList: {
        average: 1,
      },
    },
  ],
};

export const Empty = Template.bind({});
Empty.args = {};
