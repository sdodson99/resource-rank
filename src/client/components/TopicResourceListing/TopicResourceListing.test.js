import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import TopicResourceListing from './TopicResourceListing';

describe('<TopicResourceListing />', () => {
  test('it should mount', () => {
    render(<TopicResourceListing />);

    const topicResourceListing = screen.getByTestId('TopicResourceListing');

    expect(topicResourceListing).toBeInTheDocument();
  });

  describe('with multiple topic resources', () => {
    const topicResources = [
      {
        id: '123',
        resource: {
          id: '123',
        },
      },
      {
        id: '456',
        resource: {
          id: '123',
        },
      },
      {
        id: '789',
        resource: {
          id: '123',
        },
      },
    ];

    it('should render correctly', () => {
      const tree = renderer
        .create(
          <TopicResourceListing topicId={'123'} topics={topicResources} />
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('without topic resources', () => {
    it('should render correctly', () => {
      const tree = renderer
        .create(<TopicResourceListing topicId={'123'} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
