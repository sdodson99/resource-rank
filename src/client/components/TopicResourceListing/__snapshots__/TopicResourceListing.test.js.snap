// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopicResourceListing /> with multiple topic resources should render correctly 1`] = `
<div
  className="TopicResourceListing"
  data-testid="TopicResourceListing"
>
  <div
    className="border-b"
  >
    <div
      className="TopicResourceListingItem"
      data-testid="TopicResourceListingItem"
    >
      <a
        href="/topics/topic-name/resources/resource-name"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="ListingItem HoverListingItem"
          data-testid="ListingItem"
        >
          <div
            className="sm:flex sm:justify-between items-center sm:text-left text-center"
          >
            <div
              className="col-sm-auto flex flex-col items-center sm:items-start"
            >
              <div
                className="flex items-center justify-center sm:justify-start"
              >
                <div />
              </div>
              <div
                className="mt-1"
              >
                <div
                  className="RatingStarGroup"
                  data-testid="RatingStarGroup"
                >
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="FullRatingStar"
                        data-testid="FullRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="orange"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="EmptyRatingStar"
                        data-testid="EmptyRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="#d6d6d6"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="EmptyRatingStar"
                        data-testid="EmptyRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="#d6d6d6"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="EmptyRatingStar"
                        data-testid="EmptyRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="#d6d6d6"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="EmptyRatingStar"
                        data-testid="EmptyRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="#d6d6d6"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="mt-5 sm:mt-0"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                />
              </svg>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    className="border-b"
  >
    <div
      className="TopicResourceListingItem"
      data-testid="TopicResourceListingItem"
    >
      <a
        href="/topics/topic-name/resources/resource-name2"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="ListingItem HoverListingItem"
          data-testid="ListingItem"
        >
          <div
            className="sm:flex sm:justify-between items-center sm:text-left text-center"
          >
            <div
              className="col-sm-auto flex flex-col items-center sm:items-start"
            >
              <div
                className="flex items-center justify-center sm:justify-start"
              >
                <div />
              </div>
              <div
                className="mt-1"
              >
                <div
                  className="RatingStarGroup"
                  data-testid="RatingStarGroup"
                >
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="FullRatingStar"
                        data-testid="FullRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="orange"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="FullRatingStar"
                        data-testid="FullRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="orange"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="FullRatingStar"
                        data-testid="FullRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="orange"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="EmptyRatingStar"
                        data-testid="EmptyRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="#d6d6d6"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="EmptyRatingStar"
                        data-testid="EmptyRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="#d6d6d6"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="mt-5 sm:mt-0"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                />
              </svg>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    className=""
  >
    <div
      className="TopicResourceListingItem"
      data-testid="TopicResourceListingItem"
    >
      <a
        href="/topics/topic-name/resources/resource-name3"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="ListingItem HoverListingItem"
          data-testid="ListingItem"
        >
          <div
            className="sm:flex sm:justify-between items-center sm:text-left text-center"
          >
            <div
              className="col-sm-auto flex flex-col items-center sm:items-start"
            >
              <div
                className="flex items-center justify-center sm:justify-start"
              >
                <div />
              </div>
              <div
                className="mt-1"
              >
                <div
                  className="RatingStarGroup"
                  data-testid="RatingStarGroup"
                >
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="FullRatingStar"
                        data-testid="FullRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="orange"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="FullRatingStar"
                        data-testid="FullRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="orange"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="FullRatingStar"
                        data-testid="FullRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="orange"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="FullRatingStar"
                        data-testid="FullRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="orange"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    data-testid="RatingStarGroup_RatingStar"
                    onClick={[Function]}
                  >
                    <div
                      className="RatingStar"
                      data-testid="RatingStar"
                    >
                      <div
                        className="FullRatingStar"
                        data-testid="FullRatingStar"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          color="orange"
                          focusable="false"
                          style={
                            Object {
                              "fontSize": 25,
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="mt-5 sm:mt-0"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                />
              </svg>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`<TopicResourceListing /> without topic resources should render correctly 1`] = `
<div
  className="TopicResourceListing"
  data-testid="TopicResourceListing"
/>
`;
